x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -451,33 +451,48 @@ Entry<CacheFileRegion> get(KeyType cacheKey, long fileLength, int region) {
      * @param writer    a writer that handles writing of newly downloaded data to the shared cache
      * @param listener  listener that is called once all downloading has finished
      *
-     * @return {@code true} if there were enough free pages to start downloading
+     * @return {@code true} if there were enough free pages to start downloading the full entry
      */
     public boolean maybeFetchFullEntry(KeyType cacheKey, long length, RangeMissingHandler writer, ActionListener<Void> listener) {
         int finalRegion = getEndingRegion(length);
         if (freeRegionCount() < finalRegion) {
             // Not enough room to download a full file without evicting existing data, so abort
+            listener.onResponse(null);
```

Comment:
Tiniest of nits: should this be '0' rather than 'null', like it is everywhere else?

Reply:
Nah this is the overall listener for the fetch operation, not the per-chunk one, only the per chunk one takes an int :)